components/raftstore/src/coprocessor/region_info_accessor.rs

```diff
@@ -217,17 +241,37 @@ impl RoleObserver for RegionEventListener {
     }
 }
 
+impl RegionHeartbeatObserver for RegionEventListener {
+    fn on_region_heartbeat(&self, context: &mut ObserverContext<'_>, region_stat: &RegionStat) {
+        let region = context.region().clone();
+        let region_stat = region_stat.clone();
+        let event = RaftStoreEvent::UpdateRegionActivity {
+            region,
+            activity: RegionActivity { region_stat },
+        };
+
+        self.scheduler
+            .schedule(RegionInfoQuery::RaftStoreEvent(event))
+            .unwrap();
+    }
+}
+
 /// Creates an `RegionEventListener` and register it to given coprocessor host.
 fn register_region_event_listener(
     host: &mut CoprocessorHost<impl KvEngine>,
     scheduler: Scheduler<RegionInfoQuery>,
+    enable_region_stats_manager: bool,
 ) {
     let listener = RegionEventListener { scheduler };
 
     host.registry
         .register_role_observer(1, BoxRoleObserver::new(listener.clone()));
     host.registry
-        .register_region_change_observer(1, BoxRegionChangeObserver::new(listener));
+        .register_region_change_observer(1, BoxRegionChangeObserver::new(listener.clone()));
+    if enable_region_stats_manager {
```

Comment:
well, there's no propagation for config change to here -- unless RegionInfoAccessor is re-created. (It also depends on compile flag.) I suspect this may change in the future. Any thoughts?

Reply:
done.